app/api/access_codes.py

```diff
@@ -118,70 +141,56 @@ def query(self, view_kwargs):
             query_ = AccessCode.query.filter(AccessCode.tickets.any(id=ticket.id))
         return query_
 
-    def before_post(self, args, kwargs, data):
-        """
-        method to add user_id to view_kwargs before post
-        :param args:
-        :param kwargs:
-        :param data:
-        :return:
-        """
-        kwargs['user_id'] = current_identity.id
-
-    def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            data['event_id'] = ticket.event_id
-        if view_kwargs.get('event_id'):
-            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-            data['event_id'] = event.id
-        elif view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
-            data['event_id'] = event.id
-        data['user_id'] = current_identity.id
-
-    def after_create_object(self, obj, data, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            ticket.access_codes.append(obj)
-            self.session.commit()
-
     view_kwargs = True
     decorators = (jwt_required, )
+    methods = ['GET', ]
     schema = AccessCodeSchema
     data_layer = {'session': db.session,
                   'model': AccessCode,
                   'methods': {
                       'query': query,
-                      'before_create_object': before_create_object,
-                      'after_create_object': after_create_object
                   }}
 
 
 class AccessCodeDetail(ResourceDetail):
     """
     AccessCode detail by id
     """
-    def before_get_object(self, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            if ticket.access_code_id:
-                view_kwargs['id'] = ticket.access_code_id
-            else:
-                view_kwargs['id'] = None
+    @jwt_required
+    def before_get(self, args, kwargs):
```

Comment:
@enigmaeth If the user is coorganizer level, the access-code should be accessible by id, otherwise they should not be accessible by id, as quoted in the handbook.
> Only valid and non-exhausted ones and accessed via the access code itself (not ID).

please cover the above point in a separate follow-up issue, for now remove the above and use `is_coorganizer` permission for get endpooint also.